client/server/server.go

```diff
@@ -587,13 +585,22 @@ func (s *Server) Up(callerCtx context.Context, _ *proto.UpRequest) (*proto.UpRes
 	s.statusRecorder.UpdateManagementAddress(s.config.ManagementURL.String())
 	s.statusRecorder.UpdateRosenpass(s.config.RosenpassEnabled, s.config.RosenpassPermissive)
 
-	runningWg := sync.WaitGroup{}
-	runningWg.Add(1)
-	go s.connectWithRetryRuns(ctx, s.config, s.statusRecorder, &runningWg)
-
-	runningWg.Wait()
+	runningChan := make(chan error)
+	go s.connectWithRetryRuns(ctx, s.config, s.statusRecorder, runningChan)
 
-	return &proto.UpResponse{}, nil
+	for {
+		select {
+		case err := <-runningChan:
+			if err != nil {
+				log.Debugf("waiting for engine to become ready failed: %s", err)
```

Comment:
Shouldn't this return the error?

Reply:
No we do have the silent retry. If I would return this would break. I am just gonna log here. This is just a channel for us to get the errors outside.

Reply:
Got it